packages/lesswrong/components/common/withRecordPostView.tsx

```diff
@@ -39,6 +39,9 @@ export const useRecordPostView = (post: PostsBase): {recordPostView: any, isRead
       // a post id has been found & it's has not been seen yet on this client session
       if (!postsRead[post._id]) {
 
+        // Update the client-side read status cache
+        setPostRead(post._id, true);
+
```

Comment:
Previously the local state was being updated after `increasePostViewCount` resolved, which caused a noticeable delay.  Need to figure out whether moving this up causes any undesirable behaviors.